### PR TITLE
MGDOBR-188 Create kustomize overlays for basic usage scenarios

### PR DESCRIPTION
**Pull request by sutaakar**
Signed-off-by: Karel Suta <ksuta@redhat.com>

[MGDOBR-188](https://issues.redhat.com/browse/MGDOBR-188) 

Refactored kustomize script to support deploying all components into local Minikube cluster.

Pending tasks:
- Publish minikube operator image?

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by sutaakar** [comment]
Currently covering only operator scenarios, will extend it to cover manager too.

**Comment by sutaakar** [comment]
@r00ta The main part of the PR is ready for review. Please take a look when you will have time.
There are some duplications of resources from dev folder.

Thinking about publishing minikube image for operator together with OpenShift one, so people can use it in kustomize directly, what do you think?

**Comment by r00ta** [comment]
Hey @sutaakar , I see that all the configurations are in the `base-..` folder and the `overlays` are just including/excluding some of the resources. 
Actually, the idea behind this kustomize project was to keep the `base` as the base layer, and then the `overlay` as the specification for each environment (for example, you can see that the `base` does not create any Openshift Route, because it is created in the `prod` overlay). 

Now, regarding the deployment of the entire platform to kubernetes, I wish it is possible to do it just creating a new `kubernetes` overlay and specify the resources there (ingresses, etc..).

I see you'd like to use kustomize also to setup just kafka/keycloak for local development, but I think we might do that in the `dev` folder and leave this `kustomize` folder just for the deployment of the entire platform to a particular environment. 

What do you think? If you have a different idea, let's talk about it ;)

**Comment by sutaakar** [comment]
@r00ta I used this approach also to simplify deployment for local development. As I see now there are several steps needed to start the local environment. By adding it into kustomize it can be created by one command (+ one more to start Keycloak).
Also it allows sharing of configuration with the complex setup.

**Comment by sutaakar** [comment]
@r00ta Could you please take a look?

**Comment by r00ta** [comment]
Looking now

**Comment by r00ta** [comment]
Nice work! I have just a minor nitpick :)

**Comment by sutaakar** [comment]
@r00ta Thanks, what nitpick?

**Review by r00ta — COMMENTED** [review]
Sorry, GH ide did not push the comment :)

File: kustomize/README.md [hunk 1]
```diff
@@ -1,6 +1,28 @@
 # Kustomization
 
-This directory contains the GitOps project used for the deployment of the platform. 
+This directory contains the GitOps project used for the deployment of the platform. Can be used to deploy platform components to support local cloud development or for demo cluster.
+
+## Local cloud development
```
[the remark this answers is not in the document]
Comment (sutaakar) [reply kustomize/README.md:5]: Sure, fine for me

**Comment by sutaakar** [comment]
@r00ta Done

**Review by r00ta — APPROVED** [review]

**Review by r00ta — APPROVED** [review]

**Review by r00ta — APPROVED** [review]

**Pull request merged**